mailbox/jpa/src/main/java/org/apache/james/mailbox/jpa/user/JPASubscriptionMapper.java

```diff
@@ -51,17 +50,11 @@ public JPASubscriptionMapper(EntityManagerFactory entityManagerFactory) {
     @Override
     public void save(Subscription subscription) throws SubscriptionException {
         EntityManager entityManager = getEntityManager();
-        EntityTransaction transaction = entityManager.getTransaction();
-        transaction.begin();
```

Comment:
It's not ok because you introduce a temporal coupling: this method will work only if called from a method opening a transaction first.
It's the downside of being so low-level because most frameworks use a declarative @Transactional that will open a transaction if no transaction exists yet.
So, please keep the transaction here.

Reply:
I can not without reworking fully thge way our mailbox-jpa implementation works, which I won't do here.

For the record, this design allow to wrap several mapper calls into a single transaction.

I would say temporal coupling here was intended by design and I should not change that fact.

Reply:
> I can not without reworking fully thge way our mailbox-jpa implementation works, which I won't do here.

False dilemma in my opinion. It's not hard to implement "isActive() || begin()" and it's probably all that is needed to remove temporal coupling.

Reply:
- 1. That is a general problem within mailbox-jpa, not just scoped at the JPASubscriptionMapper level.

 - 2. We need to keep track within this method if a transaction was open or not, so that we know if it is our responsibility to commit/rollback or not...